File: python/examples/get_latest_lotto.py
"""Fetch the latest South African Lotto results.

    RESULTSZA_API_KEY=your_key python get_latest_lotto.py

Get an API key at https://resultsza.co.za/generate_api_key
"""
import os

from resultsza import ResultsZA

client = ResultsZA(api_key=os.environ["RESULTSZA_API_KEY"])

# Omit `date` for the most recent draw, or pass "YYYY-MM-DD" for a specific one.
latest = client.get_lotto_results()
print(latest)

# A few more one-liners:
#   client.get_sa_powerball_results()
#   client.get_euromillions_results(date="2026-07-07")
#   client.get_uk49s_results(game="UK49s Lunchtime")
#   client.generate_random_numbers(game="Powerball", lines=5)
#   client.get_hot_cold_numbers(product="Lotto")

File: node/examples/getLatestLotto.ts
/**
 * Fetch the latest South African Lotto results.
 *
 *   RESULTSZA_API_KEY=your_key npx tsx examples/getLatestLotto.ts
 *
 * Get an API key at https://resultsza.co.za/generate_api_key
 */
// In your project: import { ResultsZA } from "resultsza-sdk";
import { ResultsZA } from "../src/index";

const client = new ResultsZA({ apiKey: process.env.RESULTSZA_API_KEY! });

// Omit `date` for the most recent draw, or pass "YYYY-MM-DD" for a specific one.
const latest = await client.getLottoResults();
console.log(latest);

// A few more one-liners:
//   await client.getSaPowerballResults();
//   await client.getEuromillionsResults({ date: "2026-07-07" });
//   await client.getUk49sResults({ game: "UK49s Lunchtime" });
//   await client.generateRandomNumbers({ game: "Powerball", lines: 5 });
//   await client.getHotColdNumbers({ product: "Lotto" });
